sos/report/plugins/ceph_common.py

```diff
@@ -0,0 +1,84 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+from socket import gethostname
+
+
+class Ceph_Common(Plugin, RedHatPlugin, UbuntuPlugin):
+
+    short_desc = 'CEPH common'
+
+    plugin_name = 'ceph_common'
+    profiles = ('storage', 'virt', 'container')
+
+    containers = ('ceph-(mon|rgw|osd).*',)
+    ceph_hostname = gethostname()
+
+    packages = (
+        'ceph',
+        'ceph-mds',
+        'ceph-common',
+        'libcephfs1',
+        'ceph-fs-common',
+        'calamari-server',
+        'librados2'
+    )
+
+    services = (
+        'ceph-nfs@pacemaker',
+        'ceph-mds@%s' % ceph_hostname,
+        'ceph-mon@%s' % ceph_hostname,
+        'ceph-mgr@%s' % ceph_hostname,
+        'ceph-radosgw@*',
+        'ceph-osd@*'
+    )
+
+    # This check will enable the plugin regardless of being
+    # containerized or not
+    files = ('/etc/ceph/ceph.conf',)
+
+    def setup(self):
+        all_logs = self.get_option("all_logs")
+
+        self.add_file_tags({
+            '.*/ceph.conf': 'ceph_conf',
+            '/var/log/ceph/ceph.log.*': 'ceph_log',
```

Comment:
Just noticed - the tagging for this is in `ceph_common`, but not the collection for `/var/log/ceph/ceph.log`. However, the collection is in `ceph_mon`, `ceph_mgr`, and `ceph_mds`.

Should the collection for `/var/log/ceph/ceph.log` not be solely in `ceph_common`?

Reply:
The reason I did not put the collection of /var/log/ceph/ceph.log in ceph_common is because rgw and osd nodes do not have the /var/log/ceph/ceph.log file. But common does run on each node and the file would exist if it were to be running on mon/mgr/mds nodes, so I could put the collection of that /var/log/ceph/ceph.log in common if you think that makes more sense.